### DISCUSSION TITLE
Support could be extended to 12.0 if required

### DISCUSSION THREAD

**Discussion by meeki007** [vladmandic/human#52] [Q&A] [answered]
NodeJS

    Minimum version 14.0 which is current LTS version
    Support could be extended to 12.0 if required
    WIP: Face models are not supported in NodeJS as TensorFlow/JS for Node (tfjs-node) does not implement required function

---
I was just starting to look at making a node-red node like I did for face-api but for human.
Node-red is currently nodeJS 12.x as its Recommended release.

It took me over a month to make the node-red node for your face-api so I'm in no rush to knock out a human version if it is too much of a pain to make 12.x capable. and support for tfjs-node

But if you liked my work on the last node I'd be happy to do it for human if your into working with us nutty nodeJS guys again :P

**Comment by meeki007** [comment]
I do think node-red will be going to nodeJS version 14.0 Recommended release when they release Node-RED 2.0 in April next year.
If you just want to wait them out im cool with that too :P 

node-red currently runs on nodeJS 14 just fine and I can start work early on making a node. The only prob is most users use the install scripts that install nodeJS12 by default :( also would need tfjs-node working. 

Again you do enough.  Just in the next few months if you get froggy and want to peck at the tfjs-node issue great / else thats fine too.

**Comment by vladmandic** [comment]
i'm not using any node v14 specific features, it should work fine with node v12, i just skipped it as i didn't see the need.
most likely it will work, just need to update the project and run tests. i'll update here when done.

**Comment by vladmandic** [comment, marked as answer]
done, platform support extended to nodejs 12.
there was one incompatibility, but in irrelevant part of the code - formatting of diagnostic output on console.

**Reply by meeki007** [reply]
wonderful. Thanks m8 I'll get pecking this weekend.
